hr_job_category/models/hr.py

```diff
@@ -0,0 +1,72 @@
+
+from odoo import models, fields, api
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class HRJob(models.Model):
+    _inherit = 'hr.job'
+
+    category_ids = fields.Many2many('hr.employee.category',
+                                    'job_category_rel',
+                                    'job_id',
+                                    'category_id',
+                                    string='Associated Tags')
+
+
+class HRContract(models.Model):
+    _inherit = 'hr.contract'
+
+    @api.multi
+    def _remove_tags(self, employee_id=None, job_id=None):
+        # TODO write tags only once
+        if not employee_id or not job_id:
+            return
+        employee = self.env['hr.employee'].browse(employee_id)
+        empl_tags = employee.category_ids
+        job = self.env['hr.job'].browse(job_id)
+        _logger.debug('Removing employee tags if tag exists on contract '
+                      'job: %s', empl_tags)
+        for tag in job.category_ids:
+            if tag in empl_tags:
+                employee.write({'category_ids': [(3, tag.id)]})
+
+    @api.multi
+    def _tag_employees(self, employee_id=None, job_id=None):
+        if not employee_id or not job_id:
+            return
+        employee = self.env['hr.employee'].browse(employee_id)
+        empl_tags = employee.category_ids
+        job = self.env['hr.job'].browse(job_id)
+        for tag in job.category_ids:
+            if tag not in empl_tags:
+                _logger.debug("Adding employee tag if job tag doesn't "
+                              "exists: %s", tag.name)
+                employee.write({'category_ids': [(4, tag.id)]})
+
+    @api.model
+    def create(self, vals):
+        res = super(HRContract, self).create(vals)
+        self._tag_employees(vals.get('employee_id', False),
+                            vals.get('job_id', False))
+        return res
+
+    @api.multi
+    def write(self, vals):
+        prev_data = self.read(['job_id'])
+
+        res = super(HRContract, self).write(vals)
```

Comment:
In python3 I think no need `HRContract, self` in super but it's works well to

Reply:
travis somehow better works with self )

Reply:
@i-vyshnevska this is not something of Travis, but Python version syntax.

Reply:
@pedrobaeza sorry for so much pain from my PR, it seems ok now